public/scripts/app.js

```diff
@@ -1,7 +1,150 @@
-console.log("Sanity Check: JS is working!");
+console.log('Sanity Check');
 
-$(document).ready(function(){
-
-// your code
+function renderCliffs(cliffs) {
+  for (let i = 0; i < cliffs.length; i++) {
+    renderCliff(cliffs[i]);
+  };
+  $('.updateButton').on("click", function (e) {
+      console.log('Update button clicked');
+      var currentCliffId = $(this).closest('.cliff').data('cliff-id');
+      console.log(currentCliffId);
+      $(this).closest('.cliff').find(".cliff-info").toggle();
+      $(this).closest('.cliff').find(".update-info").toggle();
+  });
+  $('.input-info').on("submit", function (e) {
+      console.log('save button clicked');
+      var currentCliffId = $(this).find('.cliff').data('cliff-id');
+      console.log(currentCliffId);
+      handleUpdate(e, currentCliffId, this);
+  });
+  $('.deleteCliff').on("click", function (e) {
+    console.log('delete button clicked');
+    var currentCliffId = $(this).closest('.cliff').data('cliff-id');
+    handleDeleteCliff(e, currentCliffId);
+  });
+}
 
+function handleUpdate(e, id, form) {
+  e.preventDefault();
+  console.log(form);
+  console.log($(form).serialize());
+  $.ajax({
+    method: "PUT",
+    url: '/api/cliffs/' + id,
+    data: $(form).serialize(),
+    success: function() {
+      window.location = '/'
```

Comment:
😭 refreshing the page is not what you want to happen after the update is successful! You'd much rather just re-render the parts of the page that have changed, namely, the single cliff that was edited.